ui/packages/catalog-ui-search/src/main/webapp/js/model/Query.js

```diff
@@ -186,7 +186,7 @@ Query.Model = PartialAssociatedModel.extend({
         type: 'text',
         isLocal: false,
         isOutdated: false,
-        'detail-level': undefined,
+        'detail-level': null,
```

Comment:
❓ Would this change impact how existing queries (i.e. queries created before this change) are handled ? I suspect it would be best if the code is able to handle both of these falsey values

Reply:
This would have no impact on existing queries, since existing queries would have an actual detail-level that was chosen, or an `undefined` detail-level that equates to `allfields`.

Reply:
Ok, if the code i still able to handle `undefined` then we should be good to go. I'm still curious though, why was this changed to `null` to begin with? 🤔

Reply:
Since we want to automatically inject the default Result form within a new Search Form, needed a way to know when a Search form is new. Because what was happening, was that when you tried to update a Search form, and you had previously saved 'All Result' for Result form, if you had a default Result Form, it was overriding your previously saved search form 'All Result'. Hence needed to know when a Search form was new so that the default form could be automatically inject, else, it should display whatever was saved if updating a Search form.

Reply:
Can we somehow apply the default Result form when the user clicks on `+ Search Form` card instead of changing the underlying model?

Reply:
That was the first thing I thought of doing, but since it does not only apply to `+ Search Form`, but it also apply to wherever a search form is created (i.e. in workspace, when first click on "Search DDF Intrigue" or when switching from queries to new queries (from text to "basic" or "advance" search)

Reply:
ohhh ok, that makes sense 👍